### PR TITLE
[IMP]Improved speed, changed code to count attribute values

### PR DESCRIPTION
**Pull request by krunal267**
Task URL:

https://www.druidoo.io/web#id=1436&action=367&model=project.task&view_type=form&menu_id=237

### CODE REVIEW

File: website_product_attribute_filter/controllers/main.py [hunk 1]
```diff
@@ -67,11 +64,27 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
...
-                    [('attribute_line_ids.value_ids', 'in', [v.id])]
-                variant_count.update(
-                    {v.id: Product.search_count(actual_domain)})
+
+        value_ids = request.env['product.attribute.value'].search(
+            [('attribute_id', 'in', attributes_ids_all.mapped('id'))])
+        if value_ids:
+            attributes_ids = value_ids.mapped('id')
+            registry = odoo.modules.registry.Registry(request.session.db)
+            new_cr = registry.cursor()
```
Comment (ivantodorovich) [inline website_product_attribute_filter/controllers/main.py]: What's the benefict of creating a new cursor vs `request.env.cr` ?

File: website_product_attribute_filter/controllers/main.py [hunk 2]
```diff
@@ -67,11 +64,27 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
...
+            from product_product \
+            inner join product_attribute_value_product_product_rel \
+            on product_attribute_value_product_product_rel.product_product_id=
+                product_product.id
+            inner join product_template on product_template.id=
+                product_product.product_tmpl_id\
+            where sale_ok is True and is_published is True and
+                product_product.active is True and
+                product_attribute_value_id in %s \
+            group by product_attribute_value_id""", [tuple(attributes_ids)])
```
Comment (ivantodorovich) [inline website_product_attribute_filter/controllers/main.py]: This is completely ignoring the search domain / terms.
We don't want that.

Performance should be improved, but behaviour should be the same.

I get the loop could be the cause of the performance issue, but why don't you try to search the products with the search_domain, first, to get their ids, and then use them on the raw SQL? That way we wouldn't be ignoring `_get_search_domain` result

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

File: website_product_attribute_filter/controllers/main.py (hunk 2 above)
Comment (krunal267) [reply website_product_attribute_filter/controllers/main.py]: Okay @ivantodorovich 

Noted it, I will forward this to Nirav.

**Comment by ivantodorovich** [comment]
Thanks @janiniravc ,

Can you please fix travis?

```
website_product_attribute_filter/controllers/main.py:5:1: F401 'odoo' imported but unused
website_product_attribute_filter/controllers/main.py:87:13: E128 continuation line under-indented for visual indent
website_product_attribute_filter/controllers/main.py:87:35: E231 missing whitespace after ','
 ```

Can you also provide a quick video showing the speed before/after this change?

**Comment by janiniravc** [comment]
hi, @ivantodorovich I will make a video.

**Comment by janiniravc** [comment]
Hello @ivantodorovich Please find attached
website_shop_issue

**Comment by ivantodorovich** [comment]
@janiniravc 


I merged the PR, But I found a bug:

Traceback (most recent call last):
  File "/home/odoo/src/odoo/odoo/addons/base/models/ir_http.py", line 203, in _dispatch
    result = request.dispatch()
  File "/home/odoo/src/odoo/odoo/http.py", line 835, in dispatch
    r = self._call_function(**self.params)
  File "/home/odoo/src/odoo/odoo/http.py", line 346, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/src/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/src/odoo/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo/src/odoo/odoo/http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "/home/odoo/src/odoo/odoo/http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/src/user/druidoo/druidoo-addons/website_product_attribute_filter/controllers/main.py", line 91, in shop
    tuple(Product.search(domain).ids),
  File "/home/odoo/src/odoo/odoo/sql_db.py", line 148, in wrapper
    return f(self, *args, **kwargs)
  File "/home/odoo/src/odoo/odoo/sql_db.py", line 225, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: syntax error at or near ")"
LINE 15:                     AND pt.id in ()
                                           ^

This happens when the search returns 0 products.
Can you please fix it in a new PR?

**Pull request merged**